app/src/main/java/com/github/swent/echo/viewmodels/HomeScreenViewModel.kt

```diff
@@ -203,11 +206,30 @@ constructor(
 
     private fun refreshFiltersContainer() {
         filterWordList = _filtersContainer.value.searchEntry.lowercase().split(" ")
-        filterTagSet = allTagSet.filter { tag -> areWordsInTag(tag, filterWordList) }.toSet()
+        val temp = allTagSet.filter { tag -> areWordsInTag(tag, filterWordList) }
+        filterTagSet = getTagsAndSubTags(temp)
+
         updateSearchMode()
         filterEvents()
     }
 
+    private fun getTagsAndSubTags(initialTagList: List<Tag>): Set<Tag> {
+        if (initialTagList.isEmpty()) return setOf()
```

Comment:
knitpick: intsead of setOf(), maybe emptySet() would be better?

Reply:
I never used emptySet() when initializing sets (or lists with emptyList()) so to make the code uniform I think I should keep it like that tbh. If you want we could still make every initialization emptySet/List later on though